pkg/remotemetrics/mapper.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remotemetrics
+
+import (
+	"fmt"
+	"strings"
+)
+
+type namespaceMapper struct {
+	namespaces []MappedNamespace
+}
+
+// NewNamespaceMapper returns a new namespace mapper.
+func NewNamespaceMapper(namespaces ...MappedNamespace) Mapper {
+	return &namespaceMapper{
+		namespaces: namespaces,
+	}
```

Comment:
Could probably filter out the namespaces which do not need to be remapped, for the sake of efficiency.